### PR TITLE
feat(kong): add revisionHistoryLimit value

### PR DESCRIPTION
**Pull request by j796160836**
#### What this PR does / why we need it:

I want to set `revisionHistoryLimit` to a smaller number to reduce the replicasets stored in the history of the deployment.

#### Special notes for your reviewer:

I'm a contribute newbie. I don't know need serial modifications or not for `gateway-operator` or `ingress` charts if add this value.  
Please let me know if I'm doing anything wrong. Thanks.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [x] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/kong/templates/deployment.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ spec:
   replicas: {{ .Values.replicaCount }}
   {{- end }}
   {{- end }}
+  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
```
Comment (pmalek) [inline charts/kong/templates/deployment.yaml]: Since that is a deployment specific field, I believe this should be nested under `deployment`.

Comment (j796160836) [reply charts/kong/templates/deployment.yaml]: okay, I will update it.

File: charts/kong/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* Add `revisionHistoryLimit` value to set the number of replicasets wants to keep.
```
Comment (pmalek) [inline charts/kong/CHANGELOG.md]:
```suggestion
* Add `deployment.revisionHistoryLimit` to set how many old `ReplicaSet`s you want to retain.
```

File: charts/kong/values.yaml [hunk 3]
```diff
@@ -19,6 +19,10 @@ deployment:
     # Setting this to false with ingressController.enabled=true will create a
     # controller-only release.
     enabled: true
+  
+  # The number of replicasets will be retain
```
Comment (pmalek) [inline charts/kong/values.yaml]:
```suggestion
  # The number of old `ReplicaSet`s to retain.
```

**Review by pmalek — COMMENTED** [review]
Just some minor nits. Otherwise LGTM.

File: charts/kong/README.md [hunk 4]
```diff
@@ -861,6 +861,7 @@ On the Gateway release side, set either `admin.tls.client.secretName` to the nam
 | ---------------------------------- | ------------------------------------------------------------------------------------- | ------------------- |
 | namespace                          | Namespace to deploy chart resources                                                   |                     |
 | deployment.kong.enabled            | Enable or disable deploying Kong                                                      | `true`              |
+| deployment.revisionHistoryLimit    | The number of replicasets will be retain                                              | `10`                |
```
Comment (pmalek) [inline charts/kong/README.md]:
```suggestion
| deployment.revisionHistoryLimit    | The number of `ReplicaSet`s to retain.                                              | `10`                |
```

**Review by pmalek — DISMISSED** [review]

File: charts/kong/values.yaml [hunk 5]
```diff
@@ -19,6 +19,10 @@ deployment:
     # Setting this to false with ingressController.enabled=true will create a
     # controller-only release.
     enabled: true
+
```
Comment (pmalek) [inline charts/kong/values.yaml]:
```suggestion

```

**Comment by pmalek** [comment]
You'll need to update the golden tests with 

```
make test.golden.update
```

to make the CI pass.

**Review by pmalek — APPROVED** [review]

**Pull request merged**